packages/core/src/components/axes/tool-bar.ts

```diff
@@ -0,0 +1,325 @@
+		// size of overflow menu icon with background
+		const iconSize = Configuration.toolBar.iconSize;
+
+		const svg = this.getContainerSVG();
+		const { width } = DOMUtils.getSVGElementSize(
+			this.services.domUtils.getMainSVG(),
+			{
+				useAttrs: true
+			}
+		);
+
+		// overflow menu option width is 160px
+		// it's set by Carbon component
+		const overflowMenuOptionWidth = 160;
+		// no good solution to get correct ToolBar position
+		// parent x doesn't work well
+		// assume the overflowMenu button has right alignment in layout
+		this.overflowMenuX = width - overflowMenuOptionWidth;
+		this.overflowMenuY =
+			parseFloat(this.parent.node().getAttribute("y")) + iconSize;
+
+		const container = DOMUtils.appendOrSelect(svg, "svg.toolbar-container")
+			.attr("width", "100%")
+			.attr("height", Configuration.toolBar.height)
+			.attr("opacity", 1);
+
+		// clean children first
+		container.html(null);
+		// loading or empty state
+		if (isDataLoading) {
+			// put an empty rect to keep space unchanged
+			DOMUtils.appendOrSelect(container, "svg.toolbar-loading-space")
+				.append("rect")
+				.attr("height", Configuration.toolBar.height)
+				.attr("width", iconSize * 3) // value doesn't matter but can't be empty
+				.attr("opacity", 0);
+		} else {
+			const self = this;
+
+			const buttonList = [];
+			// add zoom in/out button only if zoom bar is enabled
+			if (this.isZoomBarEnabled) {
+				buttonList.push(this.getZoomInButtonConfig());
+				buttonList.push(this.getZoomOutButtonConfig());
+			}
+			buttonList.push(this.getOverFlowMenuButtonConfig());
+
+			// render buttons sequentially
+			let buttonXPosition = 0;
+			buttonList.forEach((button) => {
+				// zoom in icon and event
+				const buttonContainer = DOMUtils.appendOrSelect(
+					container,
+					`svg.${button.id}`
+				);
+				// add icon back ground rect
+				let buttonSVG = `<rect class="icon-${button.iconId}" x="${buttonXPosition}px" y="0px"
+					width="${iconSize}px" height="${iconSize}px"/>`;
+				buttonSVG += button.iconSVG(buttonXPosition);
+				buttonContainer.html(buttonSVG);
+				buttonContainer.on("click", button.clickFunction);
+				buttonXPosition += iconSize;
+			});
+
+			this.overflowMenuIcon = DOMUtils.appendOrSelect(
+				this.getContainerSVG(),
+				"rect.icon-overflowRect"
+			);
+
+			const hasOpenedOverflowMenuOptions =
+				this.overflowMenuOptions
+					.selectAll("ul.bx--overflow-menu-options--open")
+					.size() > 0;
+
+			document.body.addEventListener("click", function () {
+				// always clear menu icon hover state
+				self.setOverflowMenuIconHover(false);
+				self.services.events.dispatchEvent(
+					Events.Toolbar.HIDE_MENU_ITEMS
+				);
+			});
+
+			if (hasOpenedOverflowMenuOptions) {
+				this.setOverflowMenuIconHover(true);
+				this.overflowMenuOptions.html(this.getOverflowMenuHTML());
+			} else {
+				this.setOverflowMenuIconHover(false);
+				this.overflowMenuOptions.html(null);
+			}
+		}
+	}
+
+	getOverflowMenuHTML() {
```

Comment:
is this the right name for the method? does it get you all of the HTML needed for overflow menu?

Reply:
Yes, it will return the whole html in overflow menu, including overflow menu items.